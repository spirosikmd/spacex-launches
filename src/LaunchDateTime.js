import React from 'react';
import PropTypes from 'prop-types';

const LaunchDateTime = ({ utcDate, isTentative }) => {
  const utcString = utcDate.toUTCString();

  if (isTentative) {
    return (
      <span>
        {utcString
          .split(' ')
          .slice(0, 4)
          .join(' ')}
      </span>
    );
  }

  return <span>{utcString}</span>;
};

LaunchDateTime.propTypes = {
  utcDate: PropTypes.object.isRequired,
  isTentative: PropTypes.bool.isRequired,
};

export default React.memo(LaunchDateTime);
