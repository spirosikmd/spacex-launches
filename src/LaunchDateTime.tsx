import React from 'react';
import PropTypes from 'prop-types';

interface LaunchDateTimeProps {
  utcDate: Date;
  isTentative: boolean;
}

const LaunchDateTime = ({ utcDate, isTentative }: LaunchDateTimeProps) => {
  const utcString = utcDate.toUTCString();

  if (isTentative) {
    return (
      <span>
        {utcString
          .split(' ')
          .slice(0, 4)
          .join(' ')}
      </span>
    );
  }

  return <span>{utcString}</span>;
};

LaunchDateTime.propTypes = {
  utcDate: PropTypes.object.isRequired,
  isTentative: PropTypes.bool.isRequired,
};

export default React.memo(LaunchDateTime);
