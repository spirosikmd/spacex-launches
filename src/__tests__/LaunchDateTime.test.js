import LaunchDateTime from '../LaunchDateTime';

describe('LaunchDateTime', () => {
  let props;

  beforeEach(() => {
    props = {
      utcDate: new Date('2019-01-01'),
      isTentative: false,
    };
  });

  it('renders both date and time when launch is not tentative', () => {
    expect(shallow(LaunchDateTime, props)).toMatchSnapshot();
  });

  it('renders only date when launch is tentative', () => {
    props.isTentative = true;
    expect(shallow(LaunchDateTime, props)).toMatchSnapshot();
  });
});
