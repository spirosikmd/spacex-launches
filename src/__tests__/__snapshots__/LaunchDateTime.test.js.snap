// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchDateTime renders both date and time when launch is not tentative 1`] = `
<span>
  Fri, 11 Jan 2019 15:31:00 GMT
</span>
`;

exports[`LaunchDateTime renders only date when launch is tentative 1`] = `
<span>
  Fri, 11 Jan 2019
</span>
`;
